### PR TITLE
Reset branch

### PR DESCRIPTION
**Pull request by MisterAJ-Prodigy**
Reset to beginning. I will update the CSS as I go.

### CODE REVIEW

**Comment by MisterAJ-Prodigy** [comment]
Is there a better way the to do the global object than Declare it ahead of time like I have?

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    },
+
+    "globals": {
+        "Backbone": true,
+        "_": true,
+        "Handlebars": true
+    },
+
+    "rules": {
+        "indent": [2, 4]
```
Comment (trevorwright) [inline .eslintrc]: Out of curiosity, why both?

Comment (MisterAJ-Prodigy) [reply .eslintrc]: 2 is to enforce always and 4 is the amount of spaces

File: assets/scripts/models/Todo.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+app.Todo = Backbone.Model.extend({
+    defaults: {
+        completed: false,
+    },
+
+    validate(attrs) {
+        if (!attrs.title) return new Error('Title is Required');
+        return 'Todo Created';
```
Comment (trevorwright) [inline assets/scripts/models/Todo.js]: According to the docs, you shouldn't return anything if it is valid. If something isn't valid, you return a string with the error, not an error object.

http://backbonejs.org/#Model-validate

File: assets/scripts/models/Todo.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+app.Todo = Backbone.Model.extend({
+    defaults: {
+        completed: false,
```
Comment (trevorwright) [inline assets/scripts/models/Todo.js]: convention is to include all properties the model supports in defaults, which means you should add title here.

File: assets/scripts/views/TodoAppView.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+app.TodoAppView = Backbone.View.extend({
+    initialize(options) {
+        this.template = Handlebars.compile($('#js-todo-app-view').html());
+        this.render();
+    },
+
+    render() {
+        this.$el.html(this.template);
```
Comment (trevorwright) [inline assets/scripts/views/TodoAppView.js]: Shouldn't you invoke the template here? `this.template()`

File: assets/styles/main.css [hunk 5]
```diff
@@ -1,35 +1,67 @@
 html, body {
     width: 100%;
     height: 100%;
-    background-color: #9BC362;
+    background-image: url(../images/cork.jpg);
```
Comment (trevorwright) [inline assets/styles/main.css]: Same comment about short hand background property

File: assets/scripts/views/TodoAppView.js (hunk 4 above)
Comment (MisterAJ-Prodigy) [reply assets/scripts/views/TodoAppView.js]: Yes I should

File: assets/styles/main.css [hunk 6]
```diff
@@ -1,35 +1,67 @@
...
 }
 
 #container {
     width: 100%
 }
 
-#title {
+.todo_view__title {
```
Comment (trevorwright) [inline assets/styles/main.css]: Check last PR, there were comments about how to do BEM classes

File: index.html [hunk 7]
```diff
@@ -26,32 +26,25 @@
     <script src="https://cdnjs.cloudflare.com/ajax/libs/lodash.js/4.17.4/lodash.min.js"></script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/handlebars.js/4.0.11/handlebars.min.js"></script>
 
-    <script id="todo_template" type="text/x-handlebars-template">
-        <h1 id='title'>Aj's Magical Todo Maker!!</h1>
-        <div class="main_container">
-            <form id="todo_form">
-                <input id="todo_input"/>
+    <script id="js-todo-app-view" type="text/x-handlebars-template">
```
Comment (trevorwright) [inline index.html]: `todo-app-template` would follow the prodigy web naming conentions

File: index.html [hunk 8]
```diff
@@ -26,32 +26,25 @@
...
+        <h1 class='todo_view__title'>Aj's Magical Todo Maker!!</h1>
+        <div class="todo_view__main_container">
+            <form class="todo_view__todo_form js_todo_view__todo_form">
+                <input autofocus class="todo_view__todo_input js-todo_view__todo_input"/>
```
Comment (trevorwright) [inline index.html]: javascript classes don't follow BEM naming conventions

File: assets/scripts/models/Todo.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+app.Todo = Backbone.Model.extend({
+    defaults: {
+        title: '',
+        completed: false,
+    },
+
+    validate(attrs) {
+        if (!attrs.title) return new Error('Title is Required');
```
Comment (trevorwright) [inline assets/scripts/models/Todo.js]: Check the docs again, this shouldn't throw an error, just return a string

File: assets/scripts/views/TodoAppView.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+app.TodoAppView = Backbone.View.extend({
+    initialize(options) {
+        this.template = Handlebars.compile($('#js-todo-app-view-template').html());
```
Comment (trevorwright) [inline assets/scripts/views/TodoAppView.js]: `todo-app-template` would follow the prodigy-web standards

File: assets/styles/main.css [hunk 11]
```diff
@@ -1,35 +1,65 @@
...
-    background-color: #9BC362;
+    background: url(../images/cork.jpg);
 }
 
 #container {
     width: 100%
 }
 
-#title {
+.header__h1 {
```
Comment (trevorwright) [inline assets/styles/main.css]: `todo-main__header` would follow BEM. You generally don't directly describe your doc/css implementation in the names. Same like you wouldn't do a css class called flex container or something

File: assets/styles/main.css [hunk 12]
```diff
@@ -1,35 +1,65 @@
...
 }
 
-#todo_form {
+.todo-main__form {
     margin: auto;
+    min-width: 50%;
 }
 
-.main_container {
+.todo-main__main_container_div {
```
Comment (trevorwright) [inline assets/styles/main.css]: Same as above, might be good to read over the BEM standards again

File: assets/styles/main.css (hunk 11 above)
Comment (MisterAJ-Prodigy) [reply assets/styles/main.css]: The docs don't do a good job at conveying that

File: assets/styles/main.css [hunk 13]
```diff
@@ -1,35 +1,65 @@
...
+.todo-main__form {
+    min-width: 50%;
     margin: auto;
 }
 
-.main_container {
-    margin: 50px auto 0;
-    width: 80%;
-    background-color: #4D7514;
+.todo-main__main_container {
```
Comment (trevorwright) [inline assets/styles/main.css]: This still doesn't match BEM docs... `main-container`

File: assets/styles/main.css [hunk 14]
```diff
@@ -1,35 +1,65 @@
...
+    display: flex;
+    flex-direction: column-reverse;
+    margin: 25px auto;
+    width: 50%;
+    min-height: 50px;
+    border-radius: 20px;
+    padding-bottom: 25px;
+}
+
+.todo__list_item_container {
```
Comment (trevorwright) [inline assets/styles/main.css]: same as above

File: assets/styles/main.css [hunk 15]
```diff
@@ -1,35 +1,65 @@
...
+    width: 80%;
+    height: 100px;
+    margin: 5px auto 5px;
+    border-radius: 10px;
+    background: url(../images/paper.jpg) cover no-repeat;
+    text-align: center;
+}
+
+.todo__todo_title {
```
Comment (trevorwright) [inline assets/styles/main.css]: same as above

**Pull request merged**